src/compressor/lz4/CompressionPluginLZ4.h

```diff
@@ -33,6 +34,12 @@ class CompressionPluginLZ4 : public CompressionPlugin {
       LZ4Compressor *interface = new LZ4Compressor();
       compressor = CompressorRef(interface);
     }
+#ifdef HAVE_QATZIP
```

Comment:
can we move this `#ifdef` block into the above `if` block? as i think it's part of the initialization of the compressor, and once the compressor is initialized, so we don't need to setup it every time when we call the factory method.